sw/device/lib/testing/keymgr_testutils.c

```diff
@@ -37,3 +37,41 @@ void keymgr_testutils_check_state(const dif_keymgr_t *keymgr,
              exp_state);
   }
 }
+
+void keymgr_testutils_generate_identity(const dif_keymgr_t *keymgr) {
+  CHECK_DIF_OK(dif_keymgr_generate_identity_seed(keymgr));
+
+  while (true) {
+    dif_keymgr_status_codes_t status;
+
+    CHECK_DIF_OK(dif_keymgr_get_status_codes(keymgr, &status));
+    if (status == 0) {
+      LOG_INFO("Generating identity seed");
+    } else if (status == kDifKeymgrStatusCodeIdle) {
+      break;
+    } else {
+      LOG_ERROR("Unexpected status: %0x", status);
+      break;
+    }
+  }
```

Comment:
Perhaps it might bhe more natural to write...

```
status_codes_t status;
do {
  // ...
} while (status != Idle);
```

Does the unexpected branch want to be a CHECK to fail the test completely?

Reply:
sg, updated. 

> Does the unexpected branch want to be a CHECK to fail the test completely?

yes, it shouldn't return unexpected status